velero-plugins/imagestream/backup.go

```diff
@@ -31,95 +31,98 @@ func (p *BackupPlugin) AppliesTo() (velero.ResourceSelector, error) {
 
 // Execute copies local registry images into migration registry
 func (p *BackupPlugin) Execute(item runtime.Unstructured, backup *v1.Backup) (runtime.Unstructured, []velero.ResourceIdentifier, error) {
-
-	if backup.Labels[common.MigrationApplicationLabelKey] != common.MigrationApplicationLabelValue{
-		p.Log.Info("[is-backup] skipping imagestream plugin since backup is not part of CAM")
-		return item, nil, nil
+	migrationRegistry := ""
+	if backup.Labels[common.MigrationApplicationLabelKey] != common.MigrationApplicationLabelValue {
+		p.Log.Info("[is-backup] Entering imagestream plugin ")
+		tempRegistry, err := getRoute(backup.Namespace, backup.Spec.StorageLocation, common.ConfigMap)
```

Comment:
Rather than doing the temp var assignment I like Scott's idea to simple put the output of `getRoute` on an annotation so that all of the common logic remains the same.